File: cuarray/samples/cpp/src/CuArray_size.cpp
#include <cuarray.h>
#include <iostream>

int main() {
    std::cout
            << "Running "
            << __FILE__
            << std::endl;
/* Creates a new float CuArray instance */
    CuArray<float> *cuArray = new CuArray<float>();

/*
 * Initializes the CuArray with 10 rows and 5 columns
 * and allocates memory on host.
 */
    cuArray->init(10,
                  5);

/* Get the total number of values in the CuArray */
    int size = cuArray->size();

/* Print the total number of values in cuArray. */
    std::cout
            << "Number of values: "
            << size
            << std::endl;
/* Output:
 * Number of values: 50
 */

    delete cuArray;
    return 0;
}

File: cuarray/samples/python/CuArray_size.py
"""
Always precede CuArray with the data type
Here we are importing the float template.
"""
from cuarray import FloatCuArray

print("Running", __file__)

"""Create a new float CuArray instance"""
float_cuarray = FloatCuArray()

"""Initialize the float CuArray with 10 rows and 2 columns"""
float_cuarray.init(10, 2)

"""Print the total number of values in the CuArray"""
print(float_cuarray.size())